gossip/service/gossip_service.go

```diff
@@ -58,19 +58,20 @@ type GossipService interface {
 // DeliveryServiceFactory factory to create and initialize delivery service instance
 type DeliveryServiceFactory interface {
 	// Returns an instance of delivery client
-	Service(g GossipService, oac OrdererAddressConfig, msc api.MessageCryptoService) (deliverclient.DeliverService, error)
+	Service(g GossipService, oac OrdererAddressConfig, msc api.MessageCryptoService, isStaticLeader bool) (deliverclient.DeliverService, error)
 }
 
 type deliveryFactoryImpl struct {
 }
 
 // Returns an instance of delivery client
-func (*deliveryFactoryImpl) Service(g GossipService, ec OrdererAddressConfig, mcs api.MessageCryptoService) (deliverclient.DeliverService, error) {
+func (*deliveryFactoryImpl) Service(g GossipService, ec OrdererAddressConfig, mcs api.MessageCryptoService, isStaticLeader bool) (deliverclient.DeliverService, error) {
```

Comment:
Can we move the isStaticLeader bool to the deliveryFactoryImpl struct and avoid the contract changes by passing through `InitGossipService` from peer/node/start.go?

That would seem to avoid a lot of these changes...